client/common/termutils.js

```diff
@@ -84,6 +109,39 @@ export function sample_match_termvaluesetting(row, filter) {
 				}
 			} else if (t.term.type == 'survival') {
 				// don't do anything?
+			} else if (t.term.type == 'geneVariant' && t.legendFilterType == 'geneVariant_hard') {
```

Comment:
@congyu-lu I don't think a `legendFilterType` flag is necessary, the structure of the filter data should indicate how it should be handled.

Reply:
will improve in another branch